### PR TITLE
Only run OOP features if A/B experiment is enabled.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Telling @MattGertz this is continued work to break #18939 into smaller parts.

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit64_prtest please

**Comment by CyrusNajmabadi** [comment]
jenkins IO error.

**Comment by CyrusNajmabadi** [comment]
retest windows_determinism_prtest please

**Comment by CyrusNajmabadi** [comment]
nuget cache corruption

File: src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs [hunk 1]
```diff
@@ -43,6 +45,50 @@ public static Task<RemoteHostClient> TryGetRemoteHostClientAsync(this Workspace
             return clientService?.TryGetRemoteHostClientAsync(cancellationToken);
         }
 
+        public static Task<RemoteHostClient.Session> TryCreateCodeAnalysisServiceSessionAsync(
+             this Solution solution, Option<bool> option, string experimentName, CancellationToken cancellationToken)
+             => TryCreateCodeAnalysisServiceSessionAsync(solution, option, experimentName, callbackTarget: null, cancellationToken: cancellationToken);
+
+        public static bool IsOutOfProcessEnabled(this Workspace workspace, Option<bool> option, string experimentName)
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs]: Nit: Put below `TryCreateCodeAnalysisServiceSessionAsync`, and make a private non-extension method.

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs]: Will do. However, one reason it's public is that in a following PR we will be referencing this from another locatoin.   Specifically a couple of our incremental analyzers need to decide if they should actually analyze.  And the in-proc one will not analyze if all these features are actually rnning oop.

**Pull request merged**